src/methods/join.js

```diff
@@ -129,13 +130,13 @@ export class JoinImplementation {
   ): PossiblyNormalCompletion | Value {
     if (leftCompletion instanceof PossiblyNormalCompletion) {
       if (rightCompletion instanceof PossiblyNormalCompletion) {
-        this.updatePossiblyNormalCompletionWithValue(realm, rightCompletion, resultValue);
+        this.updatePossiblyNormalCompletionWithValue(realm, rightCompletion, new SimpleNormalCompletion(resultValue));
```

Comment:
Given the name of the method, it would be more appropriate to do the wrapping inside the method.